crypto/bn/bn_lib.c

```diff
@@ -414,24 +415,30 @@ BIGNUM *BN_bin2bn(const unsigned char *s, int len, BIGNUM *ret)
 /* ignore negative */
 static int bn2binpad(const BIGNUM *a, unsigned char *to, int tolen)
 {
-    int i;
+    int i, j, top;
     BN_ULONG l;
 
-    bn_check_top(a);
     i = BN_num_bytes(a);
     if (tolen == -1)
         tolen = i;
     else if (tolen < i)
         return -1;
-    /* Add leading zeroes if necessary */
-    if (tolen > i) {
-        memset(to, 0, tolen - i);
-        to += tolen - i;
+
+    if (i == 0) {
+        OPENSSL_cleanse(to, tolen);
+        return tolen;
     }
```

Comment:
I'm wondering why we have this if.

Reply:
Fuzzer crashes without it :-)

Reply:
Because a->d is still NULL?

Reply:
What do you mean "still"? It's just NULL. But zero bits doesn't have to mean a->d to be NULL...